### ISSUE TITLE
Wifite2 kills network manager

### ISSUE THREAD

**Issue by f00ch0w** [derv82/wifite2#57]
So, started the scan and Wifite kills 3 processes including network manager... which disables whole networking and it can't find any APs

Anyone has a solution to this issue?

**Comment by ritiek** [comment]
It is necessary to stop network-manager otherwise it won't let the WiFi interface stay in monitor mode. I don't think we have any other way.

**Comment by f00ch0w** [comment]
So how do I fix this? Wifite2 can't find my AP but Wifite v2 (r87) can but it's stuck on wps attack 0/0

**Comment by mtibbett67** [comment]
From https://github.com/derv82/wifite2/issues/57

Set your monitor interfaces to unmanaged.

bob79 said:
2016-04-12 08:33 AM
try "sudo gedit /etc/NetworkManager/NetworkManager.conf" and at the end add this: "[keyfile] unmanaged-devices=interface-name:wlan0mon" should eliminate any network manager conflicts. 

This helped me alot.

**Comment by f00ch0w** [comment]
Now stuck on starting captive portal access point service after selecting portal language. For the pyrit it says invalid hash file

**Comment by derv82** [comment]
I guess Wifite doesn't *have* to kill network manager... it could just spit out the result of `airmon-ng check`. I'm going to mediate on this for a bit.

**Comment by derv82** [comment]
I added a fix to *restart* network-manager when Wifite finishes: #70

Still think we need to terminate network-manager... If a solution works for *most* people, then we'll add it to Wifite.

I can add a "hack" to not kill NetworkManager in some cases -- but I'm not sure what those cases might be. It could check if the wireless interface MAC vendor is in some whitelist... but I don't know if your issue (requiring network-manager to scan for APs) is specific to Wireless cards by-vendor or by chipset or something else.

**Comment by ee1709** [comment]
A hack would be nice. I use wifite on a raspberry pi and I need the built-in wifi to stay up so I can ssh into it from my phone

**Comment by derv82** [comment]
Yikes, that's a pretty good reason to *never* kill processes unless asked.

I'll look at adding a `--kill` switch (😆) that kills these processes, otherwise show a warning that the processes may conflict with Airmon/Airodump.

**Comment by derv82** [comment]
Running wifite **without** the `--kill` switch will **not** terminate conflicting processes. But you'll see a warning, explanation, and commands to kill them yourself.

**Comment by mtibbett67** [comment]
In my kali VM I've set my wlan interfaces as unmanaged.

http://support.qacafe.com/knowledge-base/how-do-i-prevent-network-manager-from-controlling-an-interface/

Alternate KEYFILE method

An alternate method can be used to tell Network Manager to stop controlling
an interface. This is done by adding the following lines to the Network
Manager configuration file:

[main]
plugins=keyfile

[keyfile]
unmanaged-devices=mac:00:11:22:33:44:55;mac:66:77:88:99:00:aa

List the MAC address of each interface you want Network Manager to ignore,
separated with a semicolon. Make sure that MAC addresses listed here are *LOWER
CASE*.

Newer versions of NetworkManager can also use a more generic alternate
KEYFILE method that does not involve actual MAC addresses. This syntax
replaces the [keyfile] section above with the following:

[keyfile]
unmanaged-devices=interface-name:eth*,except:interface-name:eth0;interface-name:wlan*

This alows me to keep connected on the interfaces I allowed to be managed

Mark

On Mar 10, 2018 7:20 PM, "derv" <notifications@github.com> wrote:

Running wifite *without* the --kill switch will *not* terminate conflicting
processes. But you'll see a warning, explanation, and commands to kill them
yourself.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/derv82/wifite2/issues/57#issuecomment-372083840>, or mute
the thread
